src/components/ZetkinCalendar/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { Box, Button, MenuItem, TextField, Typography } from '@material-ui/core';
+import { FormattedDate, FormattedMessage as Msg } from 'react-intl';
+
+import MonthCalendar from './MonthCalendar';
+import { useFocusDate } from '../../hooks';
+import { useIntl } from 'react-intl';
+import WeekCalendar from './WeekCalendar';
+import { ZetkinCampaign, ZetkinEvent, ZetkinTask } from '../../types/zetkin';
+
+enum CALENDAR_RANGES {
+    MONTH = 'month',
+    WEEK = 'week',
+}
+
+interface ZetkinCalendarProps {
+    baseHref: string;
+    events: ZetkinEvent[];
+    campaigns: ZetkinCampaign[];
+    tasks: ZetkinTask[];
+}
+
+const ZetkinCalendar = ({ baseHref, events, campaigns , tasks }: ZetkinCalendarProps) : JSX.Element => {
+    const { orgId } = useRouter().query;
+    const intl = useIntl();
+    const { focusDate, setFocusDate } = useFocusDate();
+    const [range, setRange] = useState(CALENDAR_RANGES.MONTH);
+
+    const sortedEvents = [...events].sort((a, b) => {
+        return new Date(a.start_time).getTime() - new Date(b.start_time).getTime();
+    });
+
+    const filteredTasks = tasks.filter(task => task.deadline);
+
+    const calendars = [
+        { id: CALENDAR_RANGES.WEEK, name: intl.formatMessage({ id: 'misc.calendar.week' }) },
+        { id: CALENDAR_RANGES.MONTH, name: intl.formatMessage({ id: 'misc.calendar.month' }) },
+    ];
+
+    const handleForwardButtonClick = () => {
+        if (range === CALENDAR_RANGES.MONTH) {
+            setFocusDate(new Date(focusDate.getFullYear(), focusDate.getMonth() + 1, focusDate.getDate()));
+        }
+        else if (range === CALENDAR_RANGES.WEEK) {
+            setFocusDate(new Date(new Date(focusDate).setDate(focusDate.getDate() + 7)));
+        }
+    };
+
+    const handleBackButtonClick = () => {
+        if (range === CALENDAR_RANGES.MONTH) {
+            setFocusDate(new Date(focusDate.getFullYear(), focusDate.getMonth() - 1, focusDate.getDate()));
+        }
+        else if (range === CALENDAR_RANGES.WEEK) {
+            setFocusDate(new Date(new Date(focusDate).setDate(focusDate.getDate() - 7)));
+        }
+    };
+
+    return (
+        <Box display="flex" flexDirection="column" height={ 1 }>
+            <Box display="grid" flexGrow={ 0 } gridTemplateAreas={ `". nav view"` } gridTemplateColumns="repeat(3, minmax(0, 1fr))">
```

Comment:
this is a nice way to centre the range selector!

Reply:
Wouldn't event have occurred to me if not for you :)